packages/core/src/main/mls/MLSService/MLSService.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {
+  AddProposalArgs,
+  CommitBundle,
+  ConversationConfiguration,
+  ConversationId,
+  CoreCrypto,
+  DecryptedMessage,
+  ExternalProposalArgs,
+  ExternalProposalType,
+  ExternalRemoveProposalArgs,
+  Invitee,
+  ProposalArgs,
+  ProposalType,
+  RemoveProposalArgs,
+} from '@otak/core-crypto/platforms/web/corecrypto';
+import {APIClient} from '@wireapp/api-client';
+import {QualifiedUsers} from '../../conversation';
+import {Decoder, Encoder} from 'bazinga64';
+import type {MLSConfig} from '../types';
+
+//@todo: this function is temporary, we wait for the update from core-crypto side
+//they are returning regular array instead of Uint8Array for commit and welcome messages
+export const optionalToUint8Array = (array: Uint8Array | []): Uint8Array => {
+  return Array.isArray(array) ? Uint8Array.from(array) : array;
+};
+
+export class MLSService {
+  constructor(
+    public readonly config: MLSConfig | undefined,
+    private readonly apiClient: APIClient,
+    private readonly coreCryptoClientProvider: () => CoreCrypto | undefined,
+  ) {}
+
+  private getCoreCryptoClient() {
+    const client = this.coreCryptoClientProvider();
+    if (!client) {
+      throw new Error('Could not get coreCryptoClient');
+    }
+    return client;
+  }
+
+  public async uploadCoreCryptoCommitBundle(groupIdDecodedFromBase64: Uint8Array, commitBundle: CommitBundle) {
+    const coreCryptoClient = this.getCoreCryptoClient();
+
+    if (commitBundle.welcome) {
+      //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
+      await this.apiClient.api.conversation.postMlsWelcomeMessage(optionalToUint8Array(commitBundle.welcome));
+    }
+    if (commitBundle.commit) {
+      const messageResponse = await this.apiClient.api.conversation.postMlsMessage(
+        //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
+        optionalToUint8Array(commitBundle.commit),
+      );
+      await coreCryptoClient.commitAccepted(groupIdDecodedFromBase64);
+      return messageResponse;
+    }
+    return null;
+  }
+
+  public async addUsersToExistingMLSConversation(groupIdDecodedFromBase64: Uint8Array, invitee: Invitee[]) {
```

Comment:
I think, now that it's in `mls` scope, we can remove `MLS` from method names
```suggestion
  public async addUsersToExistingConversation(groupIdDecodedFromBase64: Uint8Array, invitee: Invitee[]) {
```